packages/core/src/createStore.ts

```diff
@@ -1,26 +1,15 @@
-import { Leaf, Tree, State, TreeId, Ctx, createCtx } from './kernel'
-import {
-  TREE,
-  noop,
-  nameToId,
-  Unit,
-  throwError,
-  getTree,
-  safetyFunc,
-  getIsAction,
-  assign,
-  getIsAtom,
-} from './shared'
-import { Action, declareAction, ActionType } from './declareAction'
-import { Atom, declareAtom, initAction, map, getState } from './declareAtom'
+import { Tree, State, TreeId, createCtx } from './kernel'
+import { throwError, getTree, safetyFunc, assign, getIsAtom } from './shared'
+import { Action } from './declareAction'
+import { Atom, declareAtom, initAction, getState } from './declareAtom'
 
 type DispatchFunction = (action: Action<any>) => any
 type SubscribeFunction = {
   <T>(target: Atom<T>, listener: (state: T) => any): () => void
   (listener: DispatchFunction): () => void
 }
 type GetStateFunction = {
-  <T>(target: Atom<T>): T
+  <T>(target: Atom<T>): T | undefined
```

Comment:
Why undefined?

Reply:
GetState always returns state of atom in store or initial state

Reply:
it's related with `getState` function that can return undefined